pkg/chunked/storage_linux.go

```diff
@@ -42,9 +43,10 @@ const (
 	containersOverrideXattr = "user.containers.override_stat"
 	bigDataKey              = "zstd-chunked-manifest"
 
-	fileTypeZstdChunked   = iota
-	fileTypeEstargz       = iota
-	fileTypeNoCompression = iota
+	fileTypeZstdChunked = iota
+	fileTypeEstargz
+	fileTypeNoCompression
+	fileTypeHole
```

Comment:
This name doesn’t make sense.

At a first glance, without actually trading down the full set of states/transitions in detail, the code should separate two concepts:
- How to decompress compressed data in this file (a function pointer, or an interface for “start decompressing and return a `Reader`” + “clean up”). Notably the `gzipReader`/`zstdReader` members fairly strongly suggest `chunkedDiffer` is inlining ~independent object types`.
- What’s the type of the current chunk (compressed, locally-available, a hole)